### PR TITLE
Todo App files for react1-week3 were added

### PR DESCRIPTION
**Pull request by Mabonye**
React1-week3 homework

### CODE REVIEW

File: react/react1/week3/vite-project/src/App.jsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import React from 'react';
+import TodoList from './components/TodoList/TodoList';
+
+function App() {
+    const [seconds, setSeconds] = useState(0);
+
+    /* 
+        Create a react timer using useEffect react hook
+    */
+    useEffect(() => {
```
Comment (LucasIversen) [inline react/react1/week3/vite-project/src/App.jsx:12]: Good

File: react/react1/week3/vite-project/src/components/TodoList/TodoList.jsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+import TodoItem from "./todoItem/todoItem";
+
+
+function TodoList(props) {
+    const [data, setData] = useState([]);
+    const [nextId, setNextId] = useState(1);
+    const [error, setError] = useState("");
+
+    const fetchData = async () => {
+        const response = await fetch("https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw");
```
Comment (LucasIversen) [inline react/react1/week3/vite-project/src/components/TodoList/TodoList.jsx]: We might want to put the fetch in a try-catch block, just to be safe :)

File: react/react1/week3/vite-project/src/components/TodoList/TodoList.jsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+            {error && <p style={{ color: 'red', fontWeight: 600 }}>{error}</p>}
+            <ul>
+                {
+                    data.map(todo => {
+                        return <TodoItem
+                            key={todo.id}
+                            description={todo.description}
+                            deadline={todo.deadline}
+                            data={data}
+                            setData={setData}
```
Comment (LucasIversen) [inline react/react1/week3/vite-project/src/components/TodoList/TodoList.jsx:100]: Nice 👍

File: react/react1/week3/vite-project/src/components/TodoList/TodoList.jsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        document.querySelector(".todo-date").value = "";
+    };
+
+
+    return (
+        <div>
+            <div>
+                <label>
+                    Todo description:
+                    <input
```
Comment (LucasIversen) [inline react/react1/week3/vite-project/src/components/TodoList/TodoList.jsx:64]: We can make the input controlled, like this

```javascript
const [description, setDescription] = useState(")

return (
  <input
    type="text"
    className="todo-description"
    name="description"
    value={description}
    onChange={(event) =>
      setDescription(event.target.value)
    }
    />
)
``` 

This way we can work with the values of the input fields much easier :) Same thing for the date input

File: react/react1/week3/vite-project/src/components/TodoList/TodoItem/TodoItem.jsx [hunk 5]
```diff
@@ -0,0 +1,100 @@
+/* eslint-disable react/prop-types */
+import React, { useState } from "react";
+import PropTypes from "prop-types";
+import './TodoItem.css';
+
+
+function TodoItem(props) {
+    const [isChecked, setIsChecked] = useState(false);
+    const [isEditing, setIsEditing] = useState(false);
+    const [description, setDescription] = useState(props.description);
```
Comment (LucasIversen) [inline react/react1/week3/vite-project/src/components/TodoList/TodoItem/TodoItem.jsx]: no need to have states on the todos. A change in the props will also make the page rerender, so we only need to change the todos in the TodoList component :)

File: react/react1/week3/vite-project/src/components/TodoList/TodoItem/TodoItem.jsx [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+                <button onClick={isEditing ? handleUpdate : handleEdit} className="edit">
+                    {editButtonText}
+                </button>
+            </div>
+
+        </li>
+    )
+}
+
+TodoItem.propTypes = {
```
Comment (LucasIversen) [inline react/react1/week3/vite-project/src/components/TodoList/TodoItem/TodoItem.jsx:94]: Nice job 💯

**Review by LucasIversen — COMMENTED** [review]
Hi Jemima. Just a few comments from me. Hope everything makes sense, if not, feel free to write me on slack. If you have a lot of other homework you need to attend to, feel free to merge as is :)

**Review by LucasIversen — COMMENTED** [review]
Hi Jemima. Just a few comments from me. Hope everything makes sense, if not, feel free to write me on slack. If you have a lot of other homework you need to attend to, feel free to merge as is :)

**Comment by Mabonye** [comment]
> Hi Jemima. Just a few comments from me. Hope everything makes sense, if not, feel free to write me on slack. If you have a lot of other homework you need to attend to, feel free to merge as is :)

Hello Lucas, 
Thanks for your time and comments, I will go through them and modify the homework:)

**Pull request merged**